portfolio/src/main/webapp/quiz.js

```diff
@@ -2,40 +2,88 @@
  * Question data type.
  */
 class Question {
-  constructor(question, answers, correct) {
+  constructor(question, answers, correctAnswerIndex) {
     this.question_ = question;
     this.answers_ = answers;
     this.correctAnswerIndex_ = correctAnswerIndex;
   }
 
   getQuestion() {
-    return question_;
+    return this.question_;
   }
 
   getAnswers() {
-    return answers_;
+    return this.answers_;
   }
 
   getCorrectAnswerIndex() {
-    return correctAnswerIndex_;
+    return this.correctAnswerIndex_;
   }
 }
 
+// create array of quiz questions 
+const questions = [];
+questions.push(
+    new Question('What\'s my GitHub username?',
+		['iolawale', 'ihsan', 'olawale314', 'ihsan314'], 3));
+questions.push(
+    new Question('Where do I reside?',
+		['San Francisco, CA', 'Vancouver, BC', 'Toronto, ON', 'Palo Alto CA'], 1));
+questions.push(
+    new Question('Where do I attend university?',
+		['University of British Columbia', 'Simon Fraser University',
+		  'Stanford University', 'New York University'], 0));
+questions.push(
+    new Question('In which of these classes did I receive the highest grade?',
+		['Principles of Microeconomics', 'Honours Integral Calculus',
+		  'Basic Algorithms and Data Structures', 'Introduction to Microcomputers'], 1));
+questions.push(
+    new Question('In which of these classes did I receive the lowest grade?',
+		['Principles of Microeconomics', 'Honours Integral Calculus',
+		  'Basic Algorithms and Data Structures', 'Introduction to Microcomputers'], 3));
+
 function presentQuiz() {
-  // create array of quiz questions 
-  const questions;
+  const questionsContainer = document.getElementById('questions-container');
+  for (let questionNum = 0; questionNum < questions.length; questionNum++) {
+    const q = questions[questionNum];
+    const answers = q.getAnswers();
+    const correctIndex = q.getCorrectAnswerIndex();
+    // Display the options
+    const questionStatement = document.createElement('p');
+    const question = document.createTextNode(`Question ${questionNum}: ${q.getQuestion()}`);
+    questionStatement.appendChild(question);
 
-  const quizContainer = document.getElementById('quiz-container');
-  const questionContainer = document.getElementById('question-container');
-  const answersContainer = document.getElementById('answers-container');
-  const submitContainer = document.getElementById('submit-container');
+    const quizArea = document.createElement('select');
+    quizArea.setAttribute('id', `question-${questionNum}`);
```

Comment:
There's some shared functionality here between `checkSubmission` and `presentQuiz` - consider abstracting this out to another function s.t. both functions can leverage the same scheme for getting the question id. The concern is that one method might drift from the other and then the quiz won't work as intended.

Reply:
I think both the form shown here in line 57 and the form in `checkSubmission` are equivalent, so this generation of the id can be abstracted safely I think.